cpusupport/cpusupport_arm_aes.c

```diff
@@ -15,6 +15,10 @@
 #endif /* __arm__ */
 #endif /* CPUSUPPORT_HWCAP_GETAUXVAL */
 
+#if defined(CPUSUPPORT_HWCAP_ELF_AUX_INFO)
+#include <sys/auxv.h>
```

Comment:
Is there a reason to not merge this with the previous block, since they both `#include <sys/auxv.h>`?

Reply:
That felt weird, since the two `<sys/auxv.h>` files are very different.  I figured it would be clearer to have two completely different sections.

For example, what if we added Solaris support for detecting cpu features?  Casual googling suggests that they don't use `<sys/auvx.h>` (if they provide that functionality at all!).  Or what if we wanted to support M1 features (if / when there's documentation about that chip)?

Ditto for having a `readcaps` macro, followed by "cross-platform" code which relied on the same `HWCAP_FOO` names.  Empirically, it looks like that would work, but I'm leery about future expandability.

I'm happy to merge the blocks as much as possible, including `readcaps`, if you think I'm being overly cautious about future expandability about this system-specific code.

Reply:
Ok, make sense.  We can reassess this if/when other platforms land similar support to see if there's enough consistency to be worth merging; I agree that it doesn't make sense to merge the code for 2 platforms if all the rest will be different.